packages/api/src/controllers/marketDirectoryInfoController.ts

```diff
@@ -84,16 +85,13 @@ const marketDirectoryInfoController = {
       res: Response,
     ) => {
       const { farm_id, ...data } = req.body;
-
+      const { id } = req.params;
+      const trx = await transaction.start(Model.knex());
       try {
         // @ts-expect-error: TS doesn't see query() through softDelete HOC; safe at runtime
-        const updated = await MarketDirectoryInfoModel.query()
-          .context({ user_id: req.auth?.user_id })
-          .findById(req.params.id)
-          .patch(data)
-          .returning('*');
-
-        return res.status(200).send(updated);
+        await baseController.upsertGraph(MarketDirectoryInfoModel, { ...data, id }, req, { trx });
```

Comment:
Check and trim string on model still seems to be working using the basecontroller function. Wondering if that function would actually be useful globally in the basecontroller.